api/tests/integration/scripts/prod/archive-campaigns_test.js

```diff
@@ -0,0 +1,59 @@
+const {
+  expect,
+  databaseBuilder,
+  knex, sinon,
+} = require('../../../test-helper');
+const archiveCampaigns = require('../../../../scripts/prod/archive-campaigns');
+
+describe('Script | Prod | Compute Badge Acquisitions', function () {
+  let clock;
+  let now;
+
+  beforeEach(function() {
+    now = new Date('2022-01-01');
+    clock = sinon.useFakeTimers(now);
+  })
+  afterEach(function () {
+    clock.restore();
+  });
+
+  describe('#archiveCampaigns', function () {
+    it('archives all given campaigns', async function () {
+      const file = './tests/tooling/fixtures/campaigns-to-archive.csv';
+      const output = { write: () => {}, cursorTo: () => {} }
+      databaseBuilder.factory.buildCampaign({ code: 'ABC123' });
+      databaseBuilder.factory.buildCampaign({ code: 'ABC456' });
+      databaseBuilder.factory.buildUser({ id: 1 });
+      databaseBuilder.factory.buildUser({ id: 2 });
+      await databaseBuilder.commit();
+
+
+      await archiveCampaigns(file, output);
+
+      const rows = await knex('campaigns').select(['code', 'archivedAt', 'archivedBy']);
+      expect(rows).to.exactlyContain([
+        { code: 'ABC123', archivedAt: now, archivedBy: 1 },
+        { code: 'ABC456', archivedAt: now, archivedBy: 2 },
+      ]);
+    });
+
+    it('archives all campaigns who do not throw an exception', async function () {
```

Comment:
**question** : ici c'est précisé "who do not throw an exception" du coup on s'attend à un autre test qui traite le cas des exceptions